Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs

```diff
@@ -72,14 +70,21 @@ private IEnumerable<ResourceClaim> GetResourceClaimLineageForResourceClaim(strin
                 throw new InvalidOperationException($"Multiple resource claims with a claim name of '{resourceClaimUri}' were found in the Ed-Fi API's security configuration. Authorization cannot be performed.", ex);
             }
 
-            if (resourceClaim != null)
+            if (resourceClaim == null)
+            {
+                return resourceClaimLineage;
+            }
+            
+            if (resourceClaimLineage.Select(rc => rc.ClaimName).Contains(resourceClaim.ClaimName))
             {
-                resourceClaimLineage.Add(resourceClaim);
+                throw new InvalidOperationException($"A cycle was detected in the resource claim hierarchy of the security metadata: '{string.Join("' -> '", resourceClaimLineage.SkipWhile(rc => rc != resourceClaim).Select(rc => rc.ClaimName))}' -> '{resourceClaimUri}'");
```

Comment:
The `SkipWhile` comparison should also be based on the `ClaimName`:
```suggestion
                throw new InvalidOperationException($"A cycle was detected in the resource claim hierarchy of the security metadata: '{string.Join("' -> '", resourceClaimLineage.Select(rc => rc.ClaimName).SkipWhile(rc => rc != resourceClaim.ClaimName))}' -> '{resourceClaimUri}'");
```